plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundReadWriteEdgeResource.java

```diff
@@ -0,0 +1,104 @@
+package edu.berkeley.ground.plugins.hive;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.URLEncoder;
+import java.util.Map;
+
+import org.apache.commons.httpclient.HttpException;
+import org.apache.commons.httpclient.HttpMethod;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.httpclient.methods.StringRequestEntity;
+import org.apache.commons.httpclient.util.HttpURLConnection;
+import org.codehaus.jackson.JsonParseException;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import com.google.gson.stream.JsonReader;
+
+import edu.berkeley.ground.api.models.Edge;
+import edu.berkeley.ground.api.models.EdgeVersion;
+import edu.berkeley.ground.api.models.Tag;
+import edu.berkeley.ground.exceptions.GroundException;
+import edu.berkeley.ground.plugins.hive.util.PluginUtil;
+
+public class GroundReadWriteEdgeResource {
+
+    public Edge getEdge(String edgeName) throws GroundException {
+        GetMethod get = new GetMethod(PluginUtil.groundServerAddress + "edges/" + edgeName);
+        return getEdge(get);
+    }
+
+    public EdgeVersion getEdgeVersion(long edgeId) throws GroundException {
+        GetMethod get = new GetMethod(PluginUtil.groundServerAddress + "edges/versions" + edgeId);
+        try {
+            return getEdgeVersion(get);
+        } catch (IOException e) {
+            throw new GroundException(e);
+        }
+    }
+
+    // create edge for input Tag
+    public Edge createEdge(String name, Map<String, Tag> tagMap) throws GroundException {
+        try {
+            String encodedUri = PluginUtil.groundServerAddress + "edges/" + URLEncoder.encode(name, "UTF-8");
+            PostMethod post = new PostMethod(encodedUri);
+            ObjectMapper objectMapper = new ObjectMapper();
+            String jsonString = objectMapper.writeValueAsString(tagMap);
+            post.setRequestEntity(PluginUtil.createRequestEntity(jsonString));
+            String response = PluginUtil.execute(post);
+            return this.constructEdge(response);
+        } catch (IOException e) {
```

Comment:
I would expect that only `PluginUtil.execute(post);` could throw an `IOException` in this context, so why enclose the whole body in a try/catch?